src/device/router_device_worker.erl

```diff
@@ -1902,9 +1909,9 @@ channel_correction_and_fopts(Packet, Region, Device, Frame, Count, ADRAdjustment
         case lists:member(link_check_req, Frame#frame.fopts) of
             true ->
                 SNR = blockchain_helium_packet_v1:snr(Packet),
-                MaxUplinkSNR = lorawan_mac_region:max_uplink_snr(
-                    list_to_binary(blockchain_helium_packet_v1:datarate(Packet))
-                ),
+                DataRateStr = list_to_binary(blockchain_helium_packet_v1:datarate(Packet)),
+                DataRateAtom = lora_plan:datarate_to_atom(DataRateStr),
```

Comment:
Is there a way we could prevent the following from bypassing the lib and going directly do `max_uplink_snr/1`?

```erlang
erlang:list_to_atom(blockchain_helium_packet_v1:datarate(Packet))
```

Going to the spec
```erlang
-spec max_uplink_snr(atom()) -> float().
```
all we see is the need of an atom.